modules/subscriptions/views.php

```diff
@@ -693,17 +694,39 @@ function jetpack_do_subscription_form( $instance ) {
 	if ( empty( $instance ) || ! is_array( $instance ) ) {
 		$instance = array();
 	}
-	$instance['show_subscribers_total'] = empty( $instance['show_subscribers_total'] ) || 'false' === $instance['show_subscribers_total'] ? false : true;
-	$show_only_email_and_button         = isset( $instance['show_only_email_and_button'] ) ? $instance['show_only_email_and_button'] : false;
+	$instance['show_subscribers_total']     = empty( $instance['show_subscribers_total'] ) || 'false' === $instance['show_subscribers_total'] ? false : true;
+	$show_only_email_and_button             = isset( $instance['show_only_email_and_button'] ) ? $instance['show_only_email_and_button'] : false;
+	$submit_button_text                     = isset( $instance['submit_button_text'] ) ? $instance['submit_button_text'] : '';
+
+	// Build up a string with the submit button's classes and styles and set it on the instance
+	$submit_button_classes_and_styles   = isset( $instance['submit_button_classes'] ) ? "class = \"{$instance['submit_button_classes']}\" " : '';
```

Comment:
This needs to be properly escaped.

Reply:
Might be better if we escaped it closer to the output though.

Reply:
This already is escaped. See lines 291 and 343 where the submit button text is escaped, and lines 707 and 711 where the classes and styles in $submit_button_classes_and_styles is escaped.

Reply:
`submit_button_classes` is not escaped though, so you can end up with wrong characters in there. But as Enej suggested, it's probably best to escape this as you output it, not before.

Reply:
we will need to separate class and styles, so we can escape them individually using `esc_attr`